paseos/communication/transmitter_model.py

```diff
@@ -0,0 +1,64 @@
+from loguru import logger
+
+
+class TransmitterModel:
+    """This class defines a simple transmitter model."""
+
+    def __init__(
+            self,
+            input_power: float,
+            power_efficiency: float,
+            antenna_efficiency: float,
+            line_losses: float,
+            point_losses: float,
+            antenna_gain: float = 0,
+            antenna_diameter: float = 0,
+    ) -> None:
+        """Initializes the model.
+
+        Args:
+            input_power (float): Input power into the signal amplifier, in W.
+            power_efficiency (float): The power efficiency of the signal amplifier, determines
+            the output power.
+            antenna_efficiency (float): The efficiency of the antenna.
+            line_losses (float): The line losses of the transmitter, in dB.
+            point_losses (float): The pointing losses of the transmitter, in dB.
+            antenna_gain (float): The gain of the antenna, either this or the diameter needs to
+            be given so that gain
+            can be determined.
+            antenna_diameter (float): The diameter of the antenna, in m. Either this or the gain
+            needs to be given.
+        """
+        assert input_power > 0, "Input power needs to be higher than 0."
+        assert 0 < power_efficiency <= 1, "Power efficiency should be between 0 and 1."
+        assert (
+                0 < antenna_efficiency <= 1
+        ), "Antenna efficiency should be between 0 and 1."
+        assert line_losses >= 0, "Line losses needs to be 0 or higher."
+        assert point_losses >= 0, "Pointing losses needs to be 0 or higher."
```

Comment:
as a general comment, I would suggest to move most input value checks into the actor_builder. this is how it is done for the rest of the code and it can save you a lot of code duplication in your constructors